### PR TITLE
WIP: Add `tracksClusterViewChanges` prop.

### PR DESCRIPTION
**Pull request by xseignard**
In order to improve cluster rendering performance, this props let the user decide wether or not the cluster marker should track the view changes.

See:
- https://github.com/react-native-community/react-native-maps/blob/master/docs/marker.md (`tracksViewChanges` prop)
- https://github.com/venits/react-native-map-clustering/issues/133#issuecomment-599028926

### CODE REVIEW

File: lib/ClusteredMapView.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ const ClusteredMapView = ({
   layoutAnimationConf,
   animationEnabled,
   renderCluster,
+  tracksClusterViewChandges,
```
Comment (WrathChaos) [inline lib/ClusteredMapView.js:31]: Wrong naming, it should be: `tracksClusterViewChanges`.

File: lib/ClusteredMapView.js [hunk 2]
```diff
@@ -168,6 +169,7 @@ const ClusteredMapView = ({
               onPress={_onClusterPress(marker)}
               clusterColor={clusterColor}
               clusterTextColor={clusterTextColor}
+              tracksClusterViewChandges={tracksClusterViewChandges}
```
Comment (WrathChaos) [inline lib/ClusteredMapView.js:172]: Wrong naming, it should be: tracksClusterViewChanges.

File: lib/ClusteredMapView.js [hunk 3]
```diff
@@ -204,6 +206,7 @@ ClusteredMapView.defaultProps = {
   animationEnabled: true,
   preserveClusterPressBehavior: false,
   layoutAnimationConf: LayoutAnimation.Presets.spring,
+  tracksClusterViewChandges: true,
```
Comment (WrathChaos) [inline lib/ClusteredMapView.js:209]: Wrong naming, it should be: tracksClusterViewChanges.

File: lib/ClusteredMarker.js [hunk 4]
```diff
@@ -8,7 +8,8 @@ const ClusteredMarker = ({
   properties,
   onPress,
   clusterColor,
-  clusterTextColor
+  clusterTextColor,
+  tracksClusterViewChandges
```
Comment (WrathChaos) [inline lib/ClusteredMarker.js:12]: Wrong naming, it should be: tracksClusterViewChanges.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "react-native-map-clustering",
-  "version": "3.1.2",
+  "name": "@mpvite/react-native-map-clustering",
```
Comment (WrathChaos) [inline package.json:2]: It should not be your naming :) Just delete the @mpvite part.

**Comment by xseignard** [comment]
@WrathChaos thanks for your acute eyes! I corrected the typo and the package name.

Also, this seems to improve perfs on android: https://github.com/venits/react-native-map-clustering/pull/150/commits/f86cd5fa8a43037b3c28f7ac1712281a4f48c193

Any comment ?

**Comment by WrathChaos** [comment]
@xseignard Lovely work, thank you :) @venits waiting for your approval and new release.

File: lib/ClusteredMarker.js [hunk 6]
```diff
@@ -19,6 +20,7 @@ const ClusteredMarker = ({
         longitude: geometry.coordinates[0],
         latitude: geometry.coordinates[1]
       }}
+      tracksViewChandges={tracksClusterViewChanges}
```
Comment (ddcanh) [inline lib/ClusteredMarker.js:23]: typo here, wrong prop name: tracksViewChandges
It should be: tracksViewChanges={tracksClusterViewChanges}

Comment (xseignard) [reply lib/ClusteredMarker.js]: Damn, I'm lame at typos! Should be fixed

**Comment by xseignard** [comment]
Not sure @venits will merge this.
If not, I will publish a forked version to npm.

**Comment by venits** [comment]
Don't worry :) I am going to release next major release next week.
I need some time to check all pull requests/issues.

PS. Great job on this pull request. Thank you 👍

**Comment by xseignard** [comment]
Good news! 👍

**Comment by WrathChaos** [comment]
Good job @xseignard :)

**Comment by rezasazesh** [comment]
Can we also please add a unique key to all markers in `ClusteredMarker.js` like below to avoid the clustered markers to get animated when view changes in ios? When you play around with it the clustered areas keep moving around and it is not a nice experience for user

````
    <Marker
      //Unique key for each marker
      key={`${geometry.coordinates[0]}_${geometry.coordinates[1]}`}
      coordinate={{
        longitude: geometry.coordinates[0],
        latitude: geometry.coordinates[1]
      }}
      style={{ zIndex: points + 1 }}
      onPress={onPress}
    >

````

**Comment by xseignard** [comment]
Hello @rezasazesh I don't reproduce the problem, but it may be related to the fact I just updated to `supercluster@^7.0.0`

Can you give a try to the interim package and see if you still have the problem ?
https://www.npmjs.com/package/@mpvite/react-native-map-clustering

Regards

**Comment by xseignard** [comment]
@rezasazesh did you have time to test with my previous comment ?

**Comment by venits** [comment]
Hi 👋
I released a new version of this module: `3.2.0`. This change is there.

Thank you for your contribution ❤️

**Comment by xseignard** [comment]
👍👌

File: lib/ClusteredMapView.js (hunk 2 above)
Comment (rezasazesh) [reply lib/ClusteredMapView.js]: Why is `tracksClusterViewChanges` assigned to 'react-native-maps' Marker? Should it not be `tracksViewChanges={tracksClusterViewChandges}`

**Comment by rezasazesh** [comment]
Why is `tracksClusterViewChanges` assigned to 'react-native-maps' Marker? Should it not be `tracksViewChanges={tracksClusterViewChanges}`

This is the code in 3.2.0

````
 <Marker
      key={`${geometry.coordinates[0]}_${geometry.coordinates[1]}`}
      coordinate={{
        longitude: geometry.coordinates[0],
        latitude: geometry.coordinates[1],
      }}
      style={{ zIndex: points + 1 }}
      onPress={onPress}
      tracksClusterViewChanges={tracksClusterViewChanges}
    >
````

**Comment by venits** [comment]
@rezasazesh yes It was a mistake. In newest version it's fixed :)

**Pull request merged**